pkg/controller/clusterautoscaler/clusterautoscaler.go

```diff
@@ -68,6 +69,95 @@ const (
+// These values should be removed and replaced in the event that one of the cloud providers creates a nodegroupset processor.
+
+// AWS cloud provider ignore labels for the autoscaler.
+const (
+	// AwsIgnoredLabelEksctlInstanceId  is a label used by eksctl to identify instances.
+	AwsIgnoredLabelEksctlInstanceId = "alpha.eksctl.io/instance-id"
+
+	// AwsIgnoredLabelEksctlNodegroupName is a label used by eksctl to identify "node group" names.
+	AwsIgnoredLabelEksctlNodegroupName = "alpha.eksctl.io/nodegroup-name"
+
+	// AwsIgnoredLabelEksNodegroup is a label used by eks to identify "node group".
+	AwsIgnoredLabelEksNodegroup = "eks.amazonaws.com/nodegroup"
+
+	// AwsIgnoredLabelK8sEniconfig is a label used by the AWS CNI for custom networking.
+	AwsIgnoredLabelK8sEniconfig = "k8s.amazonaws.com/eniConfig"
+
+	// AwsIgnoredLabelLifecycle is a label used by the AWS for spot.
+	AwsIgnoredLabelLifecycle = "lifecycle"
+
+	// AwsIgnoredLabelEbsCsiZone is a label used by the AWS EBS CSI driver as a target for Persistent Volume Node Affinity.
+	AwsIgnoredLabelEbsCsiZone = "topology.ebs.csi.aws.com/zone"
+)
+
+// IBM cloud provider ignore labels for the autoscaler.
+const (
+	// IbmcloudIgnoredLabelWorkerId is a label used by the IBM Cloud Cloud Controler Manager.
+	IbmcloudIgnoredLabelWorkerId = "ibm-cloud.kubernetes.io/worker-id"
+
+	// IbmcloudIgnoredLabelVpcBlockCsi is a label used by the IBM Cloud CSI driver as a target for Persisten Volume Node Affinity.
+	IbmcloudIgnoredLabelVpcBlockCsi = "vpc-block-csi-driver-labels"
+
+	// IbmcloudIgnoredLabelVpcInstanceId on IBM Cloud when a VPC is in use.
+	IbmcloudIgnoredLabelVpcInstanceId = "ibm-cloud.kubernetes.io/vpc-instance-id"
+)
+
+// GCP cloud provider ignore labels for the autoscaler.
+const (
+	// GceIgnoredLabelGkeZone label is used to specify the zone of the instance.
+	GceIgnoredLabelGkeZone = "topology.gke.io/zone"
+)
+
+// Azure cloud provider ignore labels for the autoscaler.
+const (
+	// AzureDiskTopologyKey is the topology key of Azure Disk CSI driver.
+	AzureDiskTopologyKey = "topology.disk.csi.azure.com/zone"
+
+	// AzureNodepoolLegacyLabel is a label specifying which Azure node pool a particular node belongs to.
+	AzureNodepoolLegacyLabel = "agentpool"
+
+	// AzureNodepoolLabel is an AKS label specifying which nodepool a particular node belongs to.
+	AzureNodepoolLabel = "kubernetes.azure.com/agentpool"
+)
+
+// Alibaba cloud provider ignore labels for the autoscaler.
+const (
+	// AlicloudIgnoredLabelCsiZone is a label used by the Alibaba Cloud CSI driver as a target for Persistent Volume Node Affinity.
+	AlicloudIgnoredLabelCsiZone = "topology.diskplugin.csi.alibabacloud.com/zone"
+)
+
+// AppendBasicIgnoreLabels appends ignore labels for specific cloud provider to the arguments
+// so the autoscaler can use these labels without the user having to input them manually.
+func appendBasicIgnoreLabels(args []string, cfg *Config) []string {
+	switch strings.ToLower(cfg.CloudProvider) {
```

Comment:
i think it would be better if we could use the consts from here https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go to use for checking the platform type.

Reply:
I think so as well. That is exactly what I proposed to you on Slack + some other things we could to do make our work easier in the future. I think it might be better for me to do that in separate PR on top of this change, wdyt?
I tried with this approach at first, but it turned out that there might be changes needed to some other parts of the CAO code base.